### PR TITLE
Adding in Synthetics API

### PR DESCRIPTION
**Pull request by apigirl**
### What does this PR do?
Adds the Synthetics API docs

### Motivation
It's going GA today!

### Preview link

https://docs-staging.datadoghq.com/kaylyn/synthetics-api/api/?lang=bash#synthetics

### Additional notes

Editors please go ahead and make updates/suggestions as changes to the markdown files

### CODE REVIEW

File: content/en/api/users/users_get_code.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: Get user
 type: apicode
-order: 26.2
+order: 27.2
```
Comment (tanguylebarzic) [inline content/en/api/users/users_get_code.md]: Don't know about these order changes, but on https://docs-staging.datadoghq.com/kaylyn/synthetics-api/api/, Synthetics doesn't appear at the right place in the menu (to follow the alphabetical order, it should be before tags, not before tracing.

File: content/en/api/synthetics/post_test.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+external_redirect: /api/#create-tests
+---
+
+## Create a test
+
+Create a Synthetics test to initiate and configure the tests you want Datadog to send to your API endpoints or to your browser app. You can configure the endpoints being tested, the number of tests, and where they are coming from.
+
+##### Arguments
+
+*   **`assertions`** - _required_ - The function where you define when you want Datadog to alert you. Assertions are defined by: `operator`, `property`, `type`, and `target`.
```
Comment (tanguylebarzic) [inline content/en/api/synthetics/post_test.md]: operator, type, property and target are part of a given assertion. Would it be possible to indent them further (similarly to what's done on https://docs.datadoghq.com/api/?lang=python#dashboards for template_variables?

Comment (tanguylebarzic) [inline content/en/api/synthetics/post_test.md]: I found the definition a bit unclear. How about `assertions`: a list of assertions on the response of the test that should pass for the test to be considered successful. Each assertion has a `type`, an `operator`, a `target`, and possibly a `property`.

File: content/en/api/synthetics/post_test.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    <td>for <code>contains</code>/<code>does not contain</code>/<code>is</code>/<code>is not</code>: String &lt;br&gt; for <code>matches</code>/<code>does not match</code>: [RegexStrings][1]</td>
+  </tr>
+  <tr>
+    <td>Body</td>
+    <td><code>contains</code>, <code>does not contain</code>, <code>is</code>, <code>is not</code>, <code>matches</code>, <code>does not match</code></td>
+    <td>for <code>contains</code>/<code>does not contain/<code>is</code>/<code>is not</code>: String &lt;br&gt; for <code>matches</code>/<code>does not match</code>: [RegexStrings][1]</td>
+  </tr>
+</table>
+
+*   **`type`** - _required_ - The API parameter that you want to be alerted on. Types of properties are `header`, `body`, `responseTime`, and `statusCode`. When you define a header, you must specify the header parameter key in the `property` parameter, and the header parameter value with the `target` parameter. For all other types, use the `target` to specify the body, the response time, and the error messages. For example, a `"type":"statusCode"` might have a `"target":403`.
```
Comment (tanguylebarzic) [inline content/en/api/synthetics/post_test.md]: Would rephrase: `type` The part of the response you want to assess. Possible types are `header`, `body`, `responseTime`, and `statusCode`.

File: content/en/api/synthetics/post_test.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+---
+
+## Create a test
+
+Create a Synthetics test to initiate and configure the tests you want Datadog to send to your API endpoints or to your browser app. You can configure the endpoints being tested, the number of tests, and where they are coming from.
+
+##### Arguments
+
+*   **`assertions`** - _required_ - The function where you define when you want Datadog to alert you. Assertions are defined by: `operator`, `property`, `type`, and `target`.
+*   **`operator`** - _required_ - The action that you want tested. Valid operators depend on the property `type` being tested.
```
Comment (tanguylebarzic) [inline content/en/api/synthetics/post_test.md]: Would rephrase to: `operator`: Defines how to compare the `target` and the actual value from the response. The list of possible operators depends on the type of assertions... + list of the valid operators per type.

File: content/en/api/synthetics/post_test.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    <td>for <code>contains</code>/<code>does not contain/<code>is</code>/<code>is not</code>: String &lt;br&gt; for <code>matches</code>/<code>does not match</code>: [RegexStrings][1]</td>
+  </tr>
+</table>
+
+*   **`type`** - _required_ - The API parameter that you want to be alerted on. Types of properties are `header`, `body`, `responseTime`, and `statusCode`. When you define a header, you must specify the header parameter key in the `property` parameter, and the header parameter value with the `target` parameter. For all other types, use the `target` to specify the body, the response time, and the error messages. For example, a `"type":"statusCode"` might have a `"target":403`.
+*   **`property`** - _optional_ - When you are setting up a `header` `type`, this is required to define the headers parameter key. Valid values are any header keys, like `Content-Type` or `Authorization`.
+*   **`target`** - _required_ - The value of the parameter you want to test. For `header`, valid values are any of the valid values for the header key that you define in `property`. For `statusCode`, valid values are valid status codes. For `responseTime`, valid values are response times.
```
Comment (tanguylebarzic) [inline content/en/api/synthetics/post_test.md]: Would rephrase to: `target`: The expected value for the assertion, eg. the expected status code, or the maximum acceptable response time.

File: content/en/api/synthetics/post_test.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+*   **`request`** - _required_ - the array where you define your API that you want Datadog to test. Make sure to include any required authentication information.
```
Comment (tanguylebarzic) [inline content/en/api/synthetics/post_test.md]: Would rephrase to `request`: An object containing all the necessary information to perform the request to your endpoint (what URL to target, which HTTP method to use, possible headers / request body`...

Comment (tanguylebarzic) [reply content/en/api/synthetics/post_test.md]: Would also try to indent method, url, timeout, headers and body so it's clear it's part of the request.

File: content/en/api/users/users_get_code.md (hunk 1 above)
Comment (apigirl) [reply content/en/api/users/users_get_code.md]: 🤦‍♀️this is such a good catch
